commit e720ad23b4dd65ca2947e846eaecf1aabdc7b555
Author: Jim Kukunas <james.t.kukunas@linux.intel.com>
Date:   Wed Jun 8 09:37:55 2016 -0400

    UBUNTU: SAUCE: Clear Linux: xor: skip benchmark/allocations for short-circuit path
    
    We're only testing 1 algorithm ...
    
    Signed-off-by: Jim Kukunas <james.t.kukunas@linux.intel.com>
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

diff --git a/crypto/xor.c b/crypto/xor.c
index 35d6b3a..253ee61 100644
--- a/crypto/xor.c
+++ b/crypto/xor.c
@@ -110,18 +110,6 @@ calibrate_xor_blocks(void)
 	struct xor_block_template *f, *fastest;
 
 	/*
-	 * Note: Since the memory is not actually used for _anything_ but to
-	 * test the XOR speed, we don't really want kmemcheck to warn about
-	 * reading uninitialized bytes here.
-	 */
-	b1 = (void *) __get_free_pages(GFP_KERNEL | __GFP_NOTRACK, 2);
-	if (!b1) {
-		printk(KERN_WARNING "xor: Yikes!  No memory available.\n");
-		return -ENOMEM;
-	}
-	b2 = b1 + 2*PAGE_SIZE + BENCH_SIZE;
-
-	/*
 	 * If this arch/cpu has a short-circuited selection, don't loop through
 	 * all the possible functions, just test the best one
 	 */
@@ -136,26 +124,39 @@ calibrate_xor_blocks(void)
 
 	if (fastest) {
 		printk(KERN_INFO "xor: automatically using best "
-				 "checksumming function:\n");
-		xor_speed(fastest);
+				 "checksumming function: %s\n",
+				 fastest->name);
 		goto out;
-	} else {
-		printk(KERN_INFO "xor: measuring software checksum speed\n");
-		XOR_TRY_TEMPLATES;
-		fastest = template_list;
-		for (f = fastest; f; f = f->next)
-			if (f->speed > fastest->speed)
-				fastest = f;
 	}
 
+
+	/*
+	 * Note: Since the memory is not actually used for _anything_ but to
+	 * test the XOR speed, we don't really want kmemcheck to warn about
+	 * reading uninitialized bytes here.
+	 */
+	b1 = (void *) __get_free_pages(GFP_KERNEL | __GFP_NOTRACK, 2);
+	if (!b1) {
+		printk(KERN_WARNING "xor: Yikes!  No memory available.\n");
+		return -ENOMEM;
+	}
+	b2 = b1 + 2*PAGE_SIZE + BENCH_SIZE;
+
+	printk(KERN_INFO "xor: measuring software checksum speed\n");
+	XOR_TRY_TEMPLATES;
+	fastest = template_list;
+	for (f = fastest; f; f = f->next)
+		if (f->speed > fastest->speed)
+			fastest = f;
+
 	printk(KERN_INFO "xor: using function: %s (%d.%03d MB/sec)\n",
 	       fastest->name, fastest->speed / 1000, fastest->speed % 1000);
 
 #undef xor_speed
 
- out:
 	free_pages((unsigned long)b1, 2);
 
+ out:
 	active_template = fastest;
 	return 0;
 }
